File: Go (Fiber)/controller/user_controller.go
package controller

import (
	"github.com/gofiber/fiber/v2"
	"github.com/jmoiron/sqlx"
	"github.com/gofiber/fiber/v2/middleware/session"
	// "github.com/jinzhu/copier"

	"Web_CRUD/dto"
	// "Web_CRUD/model"
	"Web_CRUD/repository"
	
	

	"github.com/davecgh/go-spew/spew"
	// "fmt"
	// "log"
)

func NewUserController(db *sqlx.DB, store *session.Store) *UserController {
	userRepository := repository.NewUserRepository(db)
	return &UserController{ UserRepository: userRepository, sessionStore: store }
}








type UserController struct{
	UserRepository *repository.UserRepository
	sessionStore *session.Store
}

func (u *UserController) List(c *fiber.Ctx) error {
	session, err := u.sessionStore.Get( c )
	if err != nil { return err } // Handling for session lost

	flash := session.Get( "flash_message" )
	if flash == nil { // Set default message
		flash = ""
	}
	session.Delete( "flash_message" )
	session.Save()

	users := u.UserRepository.All()
	
	// Display view
		return c.Render( "User/list", fiber.Map{
			"users": users,
			"flash": flash,
		})
}

func (u *UserController) New(c *fiber.Ctx) error {
	// Display view
		return c.Render( "User/new", fiber.Map{} )
}

func (u *UserController) Edit(c *fiber.Ctx) error {
	id := c.Params( "id" )
	user := u.UserRepository.GetByID( id )
	// spew.Dump( user.ID )

	// Display view
		return c.Render( "User/edit", fiber.Map{
			"user": user,
		})
}

func (u *UserController) Delete(c *fiber.Ctx) error {
	id := c.Params( "id" )
	user := u.UserRepository.GetByID( id )
	spew.Dump( user.ID )

	// Display view
		return c.Render( "User/delete", fiber.Map{
			"user": user,
		})
}

func (u *UserController) New_process(c *fiber.Ctx) error {
		session, err := u.sessionStore.Get( c )
		// keys := session.Keys()
		// spew.Dump( keys, err )
		// return nil


	// spew.Dump( store )
	// Get form values
		var request dto.NewUserRequest
		err = c.BodyParser( &request )
		if err != nil {
			return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid request"})
		}

	// Validate user request
		err = dto.Validate( &request )
		if err != nil {
			return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"validation_errors": err.Error()})
		}

	// Save form data into db table
		userID, err := u.UserRepository.New( request )
		spew.Dump( userID )
		if err != nil {
			session.Set( "flash_message", "<div class='alert alert-danger'>Failed to create new user record!</div>" )
			session.Save()
			return c.Redirect( "/users" )
		}
		session.Set( "flash_message", "<div class='alert alert-success'>New user record has been created successfully!</div>" )
		session.Save()
		return c.Redirect( "/users" )
}

func (u *UserController) Edit_process(c *fiber.Ctx) error {
	id := c.Params( "id" )
	user := u.UserRepository.GetByID( id )
	spew.Dump( user.ID )

	// Display view
		return c.Render( "User/delete", fiber.Map{
			"user": user,
		})
}

func (u *UserController) Delete_process(c *fiber.Ctx) error {
	id := c.Params( "id" )
	user := u.UserRepository.GetByID( id )
	spew.Dump( user.ID )

	// Display view
		return c.Render( "User/delete", fiber.Map{
			"user": user,
		})
}

File: C# (ASP.Net Core)/Controllers/UserController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Http;
using WebCRUD.Repositories;
using WebCRUD.DTOs;
using WebCRUD.Models;
using Dumpify;
using System;

namespace WebCRUD.Controllers;

public class UserController : Controller
{
	private readonly UserRepository _userRepository;

	// Inject the UserRepository through constructor injection
	public UserController(UserRepository userRepository )
	{
		_userRepository = userRepository;
	}

	[HttpGet]
	public IActionResult Index()
	{
		return LocalRedirect( "/users" );
	}

	[HttpGet]
	public async Task<IActionResult> List()
	{
		var users = await _userRepository.GetAll();
		string flashMessage = HttpContext.Session.GetString( "flash_message" );
		HttpContext.Session.Remove( "flash_message" );

		ViewBag.Flash = flashMessage;
		return View( users );
	}

	[HttpGet]
	public IActionResult New()
	{
		return View();
	}

	[HttpGet]
	public async Task<IActionResult> Edit( int Id )
	{
		var user = await _userRepository.GetByID( Id );
		return View( user );
	}

	[HttpGet]
	public async Task<IActionResult> Delete( int Id )
	{
		var user = await _userRepository.GetByID( Id );
		return View( user );
	}

	[HttpPost]
	public async Task<IActionResult> New(NewUserRequest request)
	{
		if ( !ModelState.IsValid ) return BadRequest( ModelState );

		// Insert form data into 'users' table
			var user = new UserModel
			{
				Name = request.Name,
				Email = request.Email,
				Website = request.Website
			};
			var insertID = await _userRepository.Insert( user );

		// Set flash message then redirect to 'users' table
			if( insertID <= 0 ){
				HttpContext.Session.SetString( "flash_message", "<div class='alert alert-danger'>Failed to add new user!</div>" );
				return LocalRedirect( "/users" );
			}
			HttpContext.Session.SetString( "flash_message", "<div class='alert alert-success'>The new user has been added successfully!</div>" );
			return LocalRedirect( "/users" );
	}

	[HttpPost]
	public async Task<IActionResult> Edit(EditUserRequest request)
	{
		if ( !ModelState.IsValid ) return BadRequest( ModelState );

		// Insert form data into 'users' table
			var user = new UserModel
			{
				Id = request.Id,
				Name = request.Name,
				Email = request.Email,
				Website = request.Website
			};
			var affectedRows = await _userRepository.Update( user );

		// Set flash message then redirect to 'users' table
			if( affectedRows <=0 ){
				HttpContext.Session.SetString( "flash_message", "<div class='alert alert-danger'>Failed to update user record!</div>" );
				return LocalRedirect( "/users" );
			}
			HttpContext.Session.SetString( "flash_message", "<div class='alert alert-success'>The user record has been updated successfully!</div>" );
			return LocalRedirect( "/users" );
	}
}